### ISSUE TITLE
macOS run-time failure: Assertion failed: (!(children_allow_slice || children_allow_tile)), function kvz_encoder_state_init, file encoder_state-ctors_dtors.c, line 508.

### ISSUE THREAD

**Issue by ilovezfs** [ultravideo/kvazaar#153]
This bug is due to https://github.com/ultravideo/kvazaar/commit/19d423ab295bb53270035bc8a5c59f0298346a11

If I revert it, the problem goes away.

For Homebrew, we run the following test:
```
  resource "videosample" do
    url "https://samples.mplayerhq.hu/V-codecs/lm20.avi"
    sha256 "a0ab512c66d276fd3932aacdd6073f9734c7e246c8747c48bf5d9dd34ac8b392"
  end

  test do
    # download small sample and try to encode it
    resource("videosample").stage do
      system bin/"kvazaar", "-i", "lm20.avi", "--input-res", "16x16", "-o", "lm20.hevc"
      assert File.exist? "lm20.hevc"
    end
  end
```

With both HEAD (a4a3b0496bf5cb64d2bb8c5e03bf001a13eee4a9) and 1.0.0, the test fails with this error:
```
==> /usr/local/Cellar/kvazaar/1.0.0/bin/kvazaar -i lm20.avi --input-res 16x16 -o lm20.hevc
Compiled: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3 SSE41 SSE42 AVX AVX2
Detected: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3 SSE41 SSE42 AVX AVX2
Available: avx(11) avx2(41) sse2(2) sse41(1) 
In use: avx(2) avx2(41) 
Disabling TMVP because GOP is used.
--owf=auto value set to 0.
Assertion failed: (!(children_allow_slice || children_allow_tile)), function kvz_encoder_state_init, file encoder_state-ctors_dtors.c, line 508.
```

Build log here: https://gist.github.com/ilovezfs/80dbf9879b7d504dc59b0aa65e56fabf

**Comment by ilovezfs** [comment]
Even with https://github.com/ultravideo/kvazaar/commit/19d423ab295bb53270035bc8a5c59f0298346a11 reverted, it seems the problem can still occur. It was resolved on our Sierra and Yosemite nodes, but on El Capitan:
```
==> /usr/local/Cellar/kvazaar/1.0.0/bin/kvazaar -i lm20.avi --input-res 16x16 -o lm20.hevc
Compiled: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3
Detected: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3 SSE41 SSE42
Available: sse2(2) sse41(1) 
In use: sse2(1) sse41(1) 
Disabling TMVP because GOP is used.
--owf=auto value set to 0.
Assertion failed: (!(children_allow_slice || children_allow_tile)), function kvz_encoder_state_init, file encoder_state-ctors_dtors.c, line 508.
Error: kvazaar: failed
```

**Comment by Venti-** [comment]
Ok thanks. I suspect the small resolution (16x16) might have something to do with it. Have you tried with latest master? I recently fixed a bug which would probably result in illegal bitstream with such a small resolution and WPP enabled.

**Comment by ilovezfs** [comment]
Thanks!

**Comment by ilovezfs** [comment]
@Venti- it looks like it's getting stuck.
```
bash-3.2$ /usr/local/Cellar/kvazaar/1.0.0/bin/kvazaar -i lm20.avi --input-res 16x16 -o lm20.hevc
Compiled: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3 SSE41 SSE42 AVX AVX2
Detected: INTEL, flags: MMX SSE SSE2 SSE3 SSSE3 SSE41 SSE42 AVX AVX2
Available: avx(11) avx2(41) sse2(2) sse41(1) 
In use: avx(2) avx2(41) 
Disabling TMVP because GOP is used.
--owf=auto value set to 0.
Input: lm20.avi, output: lm20.hevc
  Video size: 16x16 (input=16x16)

```
and then it just sits there not exiting.

spindump:
[spindump.txt.zip](https://github.com/ultravideo/kvazaar/files/763148/spindump.txt.zip)

**Comment by Venti-** [comment]
Did you cherrypick the fix on top of v1.0.0? Because I see ```kvz_threadqueue_waitfor->host_get_clock_service``` in that dump, and I removed the call to that in d893474babfa12494bdefeb357f495304ff78443.

**Comment by ilovezfs** [comment]
indeed on top of 1.0.0

**Comment by Venti-** [comment]
You need  d893474 as well to have it not get stuck on OSX with threads enabled.

**Comment by ilovezfs** [comment]
Yup, fix looks good with both commit on top of 1.0.0: https://github.com/Homebrew/homebrew-core/pull/8871/files

**Comment by ilovezfs** [comment]
@Venti- Thanks so much for your help! Any idea when the successor to 1.0.0 might be tagged?

**Comment by Venti-** [comment]
I'm planning to do it by next week's friday. https://github.com/ultravideo/kvazaar/milestone/3

**Comment by ilovezfs** [comment]
@Venti- cool. I've shipped 1.0.0 with the two patches for the time being!
